webapp/channels/src/components/common/hooks/useOpenPricingModal.ts

```diff
@@ -2,24 +2,33 @@
 // See LICENSE.txt for license information.
 
 import {useCallback} from 'react';
-import {useDispatch, useSelector} from 'react-redux';
+import {useSelector} from 'react-redux';
 
 import {isCurrentLicenseCloud} from 'mattermost-redux/selectors/entities/cloud';
 
 import {trackEvent} from 'actions/telemetry_actions';
-import {openModal} from 'actions/views/modals';
 
-import PricingModal from 'components/pricing_modal';
+import {TELEMETRY_CATEGORIES} from 'utils/constants';
 
-import {ModalIdentifiers, TELEMETRY_CATEGORIES} from 'utils/constants';
+import useCWSAvailabilityCheck, {CSWAvailabilityCheckTypes} from './useCWSAvailabilityCheck';
 
 export type TelemetryProps = {
     trackingLocation: string;
 }
 
-export default function useOpenPricingModal() {
-    const dispatch = useDispatch();
+export type UseOpenPricingModalReturn = {
+    openPricingModal: (telemetryProps?: TelemetryProps) => void;
+    isAirGapped: boolean;
+}
+
+export default function useOpenPricingModal(): UseOpenPricingModalReturn {
     const isCloud = useSelector(isCurrentLicenseCloud);
+    const cwsAvailability = useCWSAvailabilityCheck();
+
+    const isAirGapped = cwsAvailability === CSWAvailabilityCheckTypes.Unavailable;
+    const canAccessExternalPricing = cwsAvailability === CSWAvailabilityCheckTypes.Available ||
+                                     cwsAvailability === CSWAvailabilityCheckTypes.NotApplicable;
```

Comment:
Seems that this condition is contradictory on the above of `isAirGapped` but also `canAccessExternalPricing` if `cwsAvailability == NotApplicable`

Reply:
As mentioned earlier, the only case would be air gapped Team Edition where this would be weird. Unless you're thinking of something else